teos/src/gatekeeper.rs

```diff
@@ -258,56 +251,45 @@ impl Gatekeeper {
 
     /// Gets a map of outdated users. Outdated users are those whose subscription has expired and the renewal grace period
     /// has already passed ([expiry_delta](Self::expiry_delta)).
-    pub(crate) fn get_outdated_users(&self, block_height: u32) -> HashMap<UserId, HashSet<UUID>> {
+    pub(crate) fn get_outdated_users(&self, block_height: u32) -> Vec<UserId> {
         let registered_users = self.registered_users.lock().unwrap().clone();
         registered_users
             .into_iter()
             .filter(|(_, info)| block_height == info.subscription_expiry + self.expiry_delta)
-            .map(|(id, info)| (id, info.appointments.keys().cloned().collect()))
-            .collect()
-    }
-
-    /// Gets a set of outdated user ids.
-    pub(crate) fn get_outdated_user_ids(&self, block_height: u32) -> HashSet<UserId> {
-        self.get_outdated_users(block_height)
-            .keys()
-            .cloned()
+            .map(|user| user.0)
             .collect()
     }
 
-    /// Get a map of outdated appointments (from any user).
-    pub(crate) fn get_outdated_appointments(&self, block_height: u32) -> HashSet<UUID> {
-        HashSet::from_iter(
-            self.get_outdated_users(block_height)
-                .into_values()
-                .flatten(),
-        )
-    }
-
-    /// Deletes a collection of appointments from the users' subscriptions (from memory only)
-    /// and updates the available_slots count for the given user.
-    ///
-    /// Notice appointments are only de-linked from users, but not actually removed. This is because the [Gatekeeper]
-    /// does not actually hold any [ExtendedAppointment](crate::extended_appointment::ExtendedAppointment) data,
-    /// just references to them.
-    pub(crate) fn delete_appointments_from_memory(
-        &self,
-        appointments: &HashMap<UUID, UserId>,
-    ) -> HashMap<UserId, UserInfo> {
-        let mut updated_users = HashMap::new();
-        let mut registered_users = self.registered_users.lock().unwrap();
-
-        for (uuid, user_id) in appointments {
-            // Remove the appointment from the appointment list and update the available slots
-            if let Some(user_info) = registered_users.get_mut(user_id) {
-                if let Some(x) = user_info.appointments.remove(uuid) {
-                    user_info.available_slots += x;
-                }
-                updated_users.insert(*user_id, user_info.clone());
-            };
+    /// Deletes these appointments from the database and updates the user's information.
+    pub(crate) fn delete_appointments(&self, appointments: Vec<UUID>, refund: bool) {
+        if appointments.is_empty() {
+            return;
```

Comment:
We may want to prevent this from happening on the caller? A bool check should be way cheaper than a context switch

Reply:
Aha right.